src/System.Private.Interop/src/InteropExtensions/PInvokeMarshal.Unix.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Security;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// This PInvokeMarshal class should provide full public Marshal 
+    /// implementation for all things related to P/Invoke marshalling
+    /// </summary>
+    public partial class PInvokeMarshal
+    {
+        public static void SaveLastWin32Error()
+        {
+            s_lastWin32Error = Interop.Sys.GetErrNo();
+        }
+
+        public static void ClearLastWin32Error()
+        {
+            Interop.Sys.ClearErrNo();
+        }
+
+        private static bool IsWin32Atom(IntPtr ptr)
+        {
+            return false;
+        }
+
+        public static unsafe String PtrToStringAnsi(IntPtr ptr)
+        {
+            if (IntPtr.Zero == ptr)
+            {
+                return null;
+            }
+
+            int len = Internal.Runtime.CompilerHelpers.InteropHelpers.strlen((byte*)ptr);
+            if (len == 0)
+            {
+                return string.Empty;
+            }
+
+            return System.Text.Encoding.UTF8.GetString((byte*)ptr, len);
+        }
+
+        public static unsafe String PtrToStringAnsi(IntPtr ptr, int len)
+        {
+            if (ptr == IntPtr.Zero)
+                throw new ArgumentNullException(nameof(ptr));
+            if (len < 0)
+                throw new ArgumentException(nameof(len));
+
+            return System.Text.Encoding.UTF8.GetString((byte*)ptr, len);
+        }
+
+        public static unsafe IntPtr MemAlloc(IntPtr cb)
+        {
+            return Interop.MemAlloc((UIntPtr)(void*)cb);
+        }
+
+        public static void MemFree(IntPtr hglobal)
+        {
+            Interop.MemFree(hglobal);
```

Comment:
Same here - this should call Marshal.CoTaskMemFree

Reply:
Move all memory API into PinvokeMarshal.cs, since Windows and Unix are same code.